### ISSUE TITLE
Number.isFinite is undefined in Internet Explorer

### ISSUE THREAD

**Issue by rismisner** [google/blockly#571]
I found a bug in field_number.js, in the numberValidator function.  It works fine in Firefox and Chrome, but Internet Explorer throws an exception because the function Number.isFinite is undefined.  The global isFinite function works, although with slightly different caveats discussed here:
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite]

I'm not sure what's the purpose of the isFinite function in this context because I don't think it's possible to enter an infinite number into the number field, and the numberValidator had already used isNaN(n) to check that it's a valid number.

Here's the function in question:

```
/**
 * Ensure that only a number in the correct range may be entered.
 * @param {string} text The user's text.
 * @return {?string} A string representing a valid number, or null if invalid.
 */
Blockly.FieldNumber.prototype.numberValidator = function(text) {
  if (text === null) {
    return null;
  }
  text = String(text);
  // TODO: Handle cases like 'ten', '1.203,14', etc.
  // 'O' is sometimes mistaken for '0' by inexperienced users.
  text = text.replace(/O/ig, '0');
  // Strip out thousands separators.
  text = text.replace(/,/g, '');
  var n = parseFloat(text || 0);
  if (isNaN(n)) {
    // Invalid number.
    return null;
  }
  // Round to nearest multiple of precision.
  if (this.precision_ && Number.isFinite(n)) { // <- this is where it throws the exception
    n = Math.round(n / this.precision_) * this.precision_;
  }
  // Get the value in range.
  n = goog.math.clamp(n, this.min_, this.max_);
  return String(n);
};
```

I suspect that the clause Number.isFinite(n) should simply be removed, though I'm not sure if there may be a good reason why it needs to be there.   Another possible solution would be to fallback to the global isFinite function if the Number.isFinite is undefined, something like (Number.isFinite ? Number.isFinite(n) : isFinite(n))

**Comment by rachel-fenichel** [comment]
Actually, you can enter "Infinity" in a number field, so this check is to avoid trying to round infinity.  
However, this is still a bug.

**Comment by rachel-fenichel** [comment]
Fixed by #575
